app/controllers/users_controller.rb

```diff
@@ -0,0 +1,29 @@
+class UsersController < ApplicationController
+  def show
+    @user = current_user
+    @user.user_profile = UserProfile.where(user_id: @user.id).first_or_create do |profile|
```

Comment:
Did you mean "find_or_create" in case you specify user_id

Reply:
And why it's on method "Show", are you sure this method should create new user? Shouldn't that be on edit?